flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -488,6 +495,15 @@ private List<StreamShardHandle> getShardsOfStream(
         return shardsOfStream;
     }
 
+    private String lookupStreamArn(String streamName) throws InterruptedException {
```

Comment:
We could `synchronize` this method to prevent multiple requests in parallel

Reply:
Ok added